lib/AST/NameLookup.cpp

```diff
@@ -2423,7 +2423,15 @@ static bool isAcceptableLookupResult(const DeclContext *dc, NLOptions options,
     // Check that there is some import in the originating context that makes
     // this decl visible.
     if (!(options & NL_IgnoreMissingImports)) {
-      if (!dc->isDeclImported(decl))
+      bool isFromClangModule = false;
+      auto definingModule = decl->getModuleContextForNameLookup();
+      if (dc->getASTContext().LangOpts.EnableCXXInterop) {
+        if (definingModule->isClangHeaderImportModule())
+          isFromClangModule = true;
+      }
+      // dc->isDeclImported(decl) false even when the correct clang module is
```

Comment:
why is `dc->isDeclImported(decl)` false?

Reply:
I managed to figure out why this function returns false inadvertently and solved the underlying problem. Please take another look.